### ISSUE TITLE
Question: Memory

### ISSUE THREAD

**Issue by lukacs-leon** [sharpie7/circuitjs1#926]
How can I add a ROM or a flash memory to my circuit, or if it is not there, can it be included?

**Comment by pfalstad** [comment]
Use the Static Ram under "digital chips".

You can use it like a ROM.  Just don't write to it.  🙂

**Comment by lukacs-leon** [comment]
Thanks @pfalstad

**Comment by lukacs-leon** [comment]
How can I write data to a static RAM?

**Comment by lukacs-leon** [comment]
Under the edit, content?

**Comment by lukacs-leon** [comment]
![Képernyőfotó 2024-02-19 - 16 07 41](https://github.com/sharpie7/circuitjs1/assets/160414449/acd135c4-0dc6-4639-873d-0cab66395318)
what's on this static RAM?

**Comment by pfalstad** [comment]
yes, you can load the static ram using right click, Edit, Contents.

What's your other question?  I can't tell what's on the static RAM by looking at a picture of it.

**Comment by lukacs-leon** [comment]
what can i do with which leg?

**Comment by pfalstad** [comment]
The WE is "write enable".  That should be low (0V) when writing, high (5V) when reading.  So if you are making a ROM, that should always be 5V.

The OE is "output enable".  The RAM is enabled when low (0V), disabled when high (5V).

A0-A3 are the address pins, and D0-D3 are the data pins.

**Comment by lukacs-leon** [comment]
Thanks

**Comment by lukacs-leon** [comment]
how big is this memory? how many Bytes / Kb?

**Comment by lukacs-leon** [comment]
does it have four serial ports or can I send four bits on a parallel port?

**Comment by pfalstad** [comment]
well, it's got 4 address pins and 4 data pins so it can store 2^4 = 16 4-bit words, or 8 bytes.

I guess you could call it a parallel port.  The data bits are accessed in parallel.

**Comment by lukacs-leon** [comment]
thank you

**Comment by lukacs-leon** [comment]
could you give me an example of something when I enter something in advance on the edit and then it reads it and writes it to a scope with an out?
